### PR TITLE
Better Bounties

### PR DESCRIPTION
**Pull request by Doxxmedearly**
- rscadd: "Added new cargo bounties."
  - rscadd: "Adjusted cargo bounties. Reward amounts are semi-randomized, as is the number of required items for some bounties. The types of bounties that appear are more randomized."
  - tweak: "Altered some bounty descriptions."
  - tweak: "Any food that has bites taken out of it will no longer count for the related bounty."
  - tweak: "More types of pies, muffins, and kabobs should now count for their respective bounties."
  - tweak: "Further adjusted cargo warehouse spawns."

Alright there's a bit going on here.

New bounties- self explanatory. We need more variety, especially since the PR that (rightfully) weeded out most of the meme ones. This includes wholly new bounties, new reagent types, and even a few different sections- engineering in particular was left out by bounties.

Semi-randomization: Just to add a bit of variety, there's some randomization. Reward amounts are picked from a range, so it's not always just one flat amount. If a bounty requires multiple items (ie, 6 toolboxes), there's some randomization, too. Rewards will be adjusted by this; if you need to collect more than the required_count, the reward will increase. 

Also it was kind of super predictable that you'd ALWAYS have x types of bounties. I've only very slightly adjusted this to provide a bit more variety.

descriptions: self explanatory. 

food bounty stuff: self explanatory. i fucking hate food typepaths.

warehouse spawns: just some further adjustments after some in-round testing. NVGs and jammers were appearing far too frequently. fire extinguishers were spammed, and I added just a bit more variety.

### CODE REVIEW

**Comment by Doxxmedearly** [comment]
!review

File: code/modules/cargo/bounties/item.dm [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
-	var/list/exclude_types // Types excluded.
+	var/list/exclude_types = list() // Types excluded.
+	var/random_count = 0 //number. This will randomize required_count when initialized by picking a random amount of required_count + and - this number. at least 1 will always be required count. Leave at 0 to not randomize.
+
 
 /datum/bounty/item/New()
 	..()
 	wanted_types = typecacheof(wanted_types)
 	exclude_types = typecacheof(exclude_types)
+	if(random_count)
```
Comment (mikomyazaki) [inline code/modules/cargo/bounties/item.dm:14]:
```suggestion
	if(random_count > 0)
```

File: code/modules/cargo/bounties/science.dm [hunk 2]
```diff
@@ -90,56 +115,45 @@
...
-	reward = 3000
-	required_count = 5
+	description = "%COMPNAME has requested some power cells to fill their supply closet. Please fully charge them first."
+	reward_low = 35
+	reward_high = 45
+	required_count = 4
+	random_count = 1
 	wanted_types = list(/obj/item/cell/apc)
 
+/datum/bounty/item/science/battery/applies_to(obj/O)
```
Comment (mikomyazaki) [inline code/modules/cargo/bounties/science.dm:131]: Instead of having:

```
/datum/bounty/item/science/battery/applies_to(obj/O)
var/obj/some/type/T = O
if!(istype(T, /obj/some/type))
    return FALSE
T.logic()
more_logic(T)
```

Everywhere, we could do:

```
/datum/bounty/item/science/battery/applies_to(var/obj/some/type/O)
if(!istype(O))
    return FALSE
O.whatever()
some_logic(O)
```

So stuff is more simply typecast. 
Ofc it's equivalent and doesn't matter too much.

File: code/modules/cargo/bounty.dm [hunk 3]
```diff
@@ -43,14 +47,14 @@
...
-/datum/bounty/proc/compatible_with(other_bounty)
+/datum/bounty/proc/compatible_with(var/datum/other_bounty)
 	return TRUE
 
 /datum/bounty/proc/mark_high_priority(scale_reward = 2)
 	if(high_priority)
 		return
 	high_priority = TRUE
-	reward = round(reward * scale_reward)
+	reward = min(round(reward * scale_reward), 15000)	//15k limit
```
Comment (mikomyazaki) [inline code/modules/cargo/bounty.dm:57]: #define this limit somewhere instead.

File: code/modules/cargo/bounty.dm [hunk 4]
```diff
@@ -155,34 +159,48 @@
 
 // Called lazily at startup to populate bounties_list with random bounties.
 /datum/controller/subsystem/cargo/proc/setupBounties()
-	for(var/i = 0; i < 3; ++i)
+	for(var/i = 0; i < 3; i++)
```
Comment (mikomyazaki) [inline code/modules/cargo/bounty.dm:162]: Maybe #define the upper loop limit somewhere so it is less magic numbery.

**Comment by alsoandanswer** [comment]
IMHO, prices are far too low.
For starters, an organ...
is the same price as a cup of ramen.
I suggest to keep the prices in the 1000 and above range as a benchmark. 
Some of these are horrendously low - lungs...for 35 credits. Cup ramen is 20 credits.

In-game, credits are pretty much worthless, and this'll create even less incentive to actually do them.
Which is pretty bad, considering they aren't really done at all in the first place.
TL;DR - raise the prices.

**Comment by Doxxmedearly** [comment]
> 
> 
> IMHO, prices are far too low.
> For starters, an organ...
> is the same price as a cup of ramen.
> I suggest to keep the prices in the 1000 and above range as a benchmark.
> Some of these are horrendously low - lungs...for 35 credits. Cup ramen is 20 credits.
> 
> In-game, credits are pretty much worthless, and this'll create even less incentive to actually do them.
> Which is pretty bad, considering they aren't really done at all in the first place.
> TL;DR - raise the prices.
> 
> Good work on increasing the variety of food items accepted though.

as the new vars in the /datum/bounty state, the amounts are multiplied by 100. 25 is 2500. This is because if I just did rand(2000, 3000) we'd get very sloppy numbers like 2743. This keeps it tidy.

**Comment by mikomyazaki** [comment]
> > IMHO, prices are far too low.
> > For starters, an organ...
> > is the same price as a cup of ramen.
> > I suggest to keep the prices in the 1000 and above range as a benchmark.
> > Some of these are horrendously low - lungs...for 35 credits. Cup ramen is 20 credits.
> > In-game, credits are pretty much worthless, and this'll create even less incentive to actually do them.
> > Which is pretty bad, considering they aren't really done at all in the first place.
> > TL;DR - raise the prices.
> > Good work on increasing the variety of food items accepted though.
> 
> as the new vars in the /datum/bounty state, the amounts are multiplied by 100. 25 is 2500. This is because if I just did rand(2000, 3000) we'd get very sloppy numbers like 2743. This keeps it tidy.

Another way to do round numbers would be to just do... round(price, 100) and round to the nearest 100. The way you've done it works, although it's a bit less clear.

**Comment by Doxxmedearly** [comment]
> 
> 
> > > IMHO, prices are far too low.
> > > For starters, an organ...
> > > is the same price as a cup of ramen.
> > > I suggest to keep the prices in the 1000 and above range as a benchmark.
> > > Some of these are horrendously low - lungs...for 35 credits. Cup ramen is 20 credits.
> > > In-game, credits are pretty much worthless, and this'll create even less incentive to actually do them.
> > > Which is pretty bad, considering they aren't really done at all in the first place.
> > > TL;DR - raise the prices.
> > > Good work on increasing the variety of food items accepted though.
> > 
> > 
> > as the new vars in the /datum/bounty state, the amounts are multiplied by 100. 25 is 2500. This is because if I just did rand(2000, 3000) we'd get very sloppy numbers like 2743. This keeps it tidy.
> 
> Another way to do round numbers would be to just do... round(price, 100) and round to the nearest 100. The way you've done it works, although it's a bit less clear.

Yeah i thought about that like a day after i uploaded it. Ugh.

**Comment by Doxxmedearly** [comment]
No point is leaving in confusing shitcode. Changed it to be more readable.

File: code/modules/cargo/bounties/bot.dm [hunk 5]
```diff
@@ -1,45 +1,50 @@
...
+	reward_low = 8500
+	reward_high = 9500
 	required_count = 2
 	wanted_types = list(/mob/living/bot/secbot/ed209)
 
 /datum/bounty/item/bot/farmbot
 	name = "Farmbot"
-	description = "After a k'ois incident multiple botanists are hospitalized. Provide a farmbot to replace them."
-	reward = 2000
+	description = "After a k'ois incident multiple botanists are hospitalized. Provide a few farmbots to handle their duties while they recover."
```
Comment (Geevies) [inline code/modules/cargo/bounties/bot.dm:27]:
```suggestion
	description = "After a k'ois incident multiple botanists were hospitalized. Provide a few farmbots to handle their duties while they recover."
```

File: code/modules/cargo/bounties/reagent.dm [hunk 6]
```diff
@@ -2,6 +2,12 @@
 	var/required_volume = 10
 	var/shipped_volume = 0
 	var/datum/reagent/wanted_reagent
+	var/list/possible_descriptions = list("We're interested in testing the quality of our stations' bartenders!",
+	"We're holding an office party and need some drink variety.",
+	"We're holding a mixing competition, and you'll receive a bonus for competing.",
+	"One of our agents has had a hard day and need to unwind so badly they're willing to put up for a bounty.",
```
Comment (Doxxmedearly) [inline code/modules/cargo/bounties/reagent.dm:8]:
```suggestion
	"One of our agents has had a hard day and needs to unwind so badly they're willing to put up for a bounty.",
```

**Comment by Doxxmedearly** [comment]
someone kick travis please
